src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    const incorrectLength: boolean = this.a <= 0 || this.b <= 0 || this.c <= 0;
+    const incorrectSides: boolean = (this.a + this.b) <= this.c
+    || (this.a + this.c) <= this.b
+    || (this.c + this.b) <= this.a;
+
+    if (incorrectLength || incorrectSides) {
```

Comment:
For these two cases, you need to throw different errors
Check 3 point in the [checklist](https://github.com/mate-academy/ts_figures_classes/blob/master/checklist.md)